README.md

```diff
@@ -150,35 +152,132 @@ and if it is disabled (boolean).
 
 ### Test Helper
 
-Since `x-select` uses internal identifiers as the `value` attribute, it
-doesn't integrate with the `fillIn` test helper. But don't fret, we've built a
-test helper for you.
+`<XSelect>` 4.0 ships with an entirely new test helper that goes
+beyond just allowing you to select an option. It allows you to
+interact with your `<select>` element in all different ways. For
+example in tests if you need to assert your first options `disabled`
+or not:
+
+```javascript
+expect(xselect.options(0).isDisabled).to.equal(true);
+```
+
+Under the hood this new test helper is using a [BigTest
+Interactor](https://bigtestjs.io/guides/interactors/introduction/). Interactors
+allow you to think about how you're going to _interact_ with the DOM
+and abstract that into composable & immutable containers. Interactors
+are similar to page objects, but for components.
 
 #### Using the test helper
 
-To use the select helper in your tests you have to import the select function:
+Import the select interactor:
 
 ``` javascript
-  import { select } from "yourappname/tests/helpers/x-select";
+// you can name the import whatever you want
+import XSelect from "yourappname/tests/helpers/x-select";
 ```
 
-We support both multiselects and regular selects. To use, you
-need to specify the class on the select (or a jquery object) as the
-first argument and the rest of the arguments are the options you'd
-like to select. For example:
+At the top of your test file you need to initialize the
+interactor. This should go at the top most part of your test so it's
+available to all tests in the file. Here's an example in Qunit:
 
-```js
-//... Single select
-  select(".my-drop-down", "My Option");
-//...
+
+``` javascript
+module("Acceptance | Your Test", function(hooks) {
+  let xselect = new XSelect('.selector-for-select');
+  setupApplicationTest(hooks);
+  // ...
+});
 ```
 
-Multiselect
-```javascript
-//... Multiselect
-  select(Ember.$(".my-drop-down"), "My Option", "My Option Two", "My Option Three");
-//...
+Once you have initialized the interactor, you're ready to start
+selecting!
+
+``` javascript
+module("Acceptance | Your Test", function(hooks) {
+  let xselect = new XSelect('.selector-for-select');
+  // ...
+
+  test('Selecting an option', async (assert) => {
+    await xselect.select('Fred Flintstone');
+    // for a multiselect pass an array
+    // await xselect.select(['Fred Flintstone', 'Bob Newhart']);
+
+    // assert change happened in your app
+  });
+});
 ```
+
+You can do more than just select options with this helper.
+
+``` javascript
+module('Acceptance | Your Test', function(hooks) {
+  let xselect = new XSelect('.selector-for-select');
+  // ...
+
+  test('Selecting an option', async (assert) => {
+    await xselect.select('Fred Flintstone');
```

Comment:
I was going to mention leaving a quick comment that says something like "the interactor resolves after _interacting_ with the select, but not after any change events are handled" which is why we use `when` for the following assertion.

But then that made me think that people will pretty much require `when` whenever they use this test helper. Or we can override the default `select` method to resolve _after it is selected_. It would still throw an error when it cannot select or when the option does not get selected, and it would mean people can `expect` as usual after awaiting of `select(option)`.